code/game/machinery/hologram.dm

```diff
@@ -68,199 +95,324 @@ var/list/holopads = list()
+		var/datum/holocall/call_to_connect = locate(href_list["connectcall"])
+		if(!qdeleted(call_to_connect))
+			call_to_connect.Answer(src)
+		temp = ""
+
+	else if(href_list["disconnectcall"])
+		var/datum/holocall/call_to_disconnect = locate(href_list["disconnectcall"])
+		if(!qdeleted(call_to_disconnect))
+			call_to_disconnect.Disconnect(src)
+		temp = ""
+
+	else if(href_list["mainmenu"])
+		temp = ""
+		if(outgoing_call)
+			outgoing_call.Disconnect()
+
+	updateDialog()
+
+//do not allow AIs to answer calls or people will use it to meta the AI sattelite
 /obj/machinery/hologram/holopad/attack_ai(mob/living/silicon/ai/user)
-	if(!istype(user))
+	if (!istype(user))
 		return
 	/*There are pretty much only three ways to interact here.
 	I don't need to check for client since they're clicking on an object.
 	This may change in the future but for now will suffice.*/
 	if(user.eyeobj.loc != src.loc)//Set client eye on the object if it's not already.
 		user.eyeobj.setLoc(get_turf(src))
-	else if(!hologram)//If there is no hologram, possibly make one.
+	else if(!masters[user])//If there is no hologram, possibly make one.
 		activate_holo(user, 0)
-	else if(master == user)//If there is a hologram, remove it. But only if the user is the master. Otherwise do nothing.
-		clear_holo()
-	return
+	else//If there is a hologram, remove it.
+		clear_holo(user)
+
+/obj/machinery/hologram/holopad/process()
+	for(var/I in masters)
+		var/mob/living/master = I
+		var/mob/living/silicon/ai/AI = master
+		if(!istype(AI))
+			AI = null
+
+		if(!qdeleted(master) && !master.incapacitated() && master.client && (!AI || AI.eyeobj))//If there is an AI attached, it's not incapacitated, it has a client, and the client eye is centered on the projector.
+			if(!(stat & NOPOWER))//If the  machine has power.
+				if(AI)	//ais are range based
+					if((HOLOPAD_MODE == 0 && (get_dist(AI.eyeobj, src) <= holo_range)))
+						return 1
 
-/obj/machinery/hologram/holopad/proc/activate_holo(mob/living/silicon/ai/user, var/force = 0)
-	if(!force && user.eyeobj.loc != src.loc) // allows holopads to pass off holograms to the next holopad in the chain
+					else if(HOLOPAD_MODE == 1)
+
+						var/area/holo_area = get_area(src)
+						var/area/eye_area = get_area(AI.eyeobj)
+
+						if(eye_area != holo_area)
+							return 1
+
+					var/turf/target_turf = get_turf(AI.eyeobj)
+					var/newdir = AI.holo.dir
+					clear_holo()//If not, we want to get rid of the hologram.
+					var/obj/machinery/hologram/holopad/pad_close = get_closest_atom(/obj/machinery/hologram/holopad, holopads, AI.eyeobj)
+					if(get_dist(pad_close, AI.eyeobj) <= pad_close.holo_range)
+						if(!(pad_close.stat & NOPOWER) && !pad_close.masters[src])
+							pad_close.activate_holo(AI, 1)
+							if(pad_close.masters[src])
+								outgoing_call.hologram.forceMove(target_turf)
+								outgoing_call.hologram.dir = newdir//runtime
+
+		clear_holo(master)//If not, we want to get rid of the hologram.

```

Comment:
Based on the comment, I suspect you are missing an else statement around here...